clang/lib/CIR/CodeGen/CIRGenAtomic.cpp

```diff
@@ -526,42 +495,54 @@ static void buildAtomicCmpXchgFailureSet(
   // compile-time value.
   CGF.getBuilder().create<mlir::cir::SwitchOp>(
       FailureOrderVal.getLoc(), FailureOrderVal,
-      [&](mlir::OpBuilder &builder, mlir::Location loc,
-          mlir::OperationState &os) {
-        SmallVector<mlir::Attribute, 3> CaseAttrs;
-        SmallVector<std::unique_ptr<mlir::Region>, 3> Regions;
-
-        // default:
-        // Unsupported memory orders get generated as memory_order_relaxed,
-        // because there is no practical way to report an error at runtime.
-        buildDefaultCase(builder, CaseAttrs, Regions, loc);
-        buildAtomicCmpXchg(CGF, E, IsWeak, Dest, Ptr, Val1, Val2, Size,
-                           SuccessOrder, mlir::cir::MemOrder::Relaxed, Scope);
-
-        // case consume:
-        // case acquire:
-        // memory_order_consume is not implemented and always falls back to
-        // memory_order_acquire
-        buildDoubleMemOrderCase(
-            builder, CaseAttrs, Regions, loc, FailureOrderVal.getType(),
-            mlir::cir::MemOrder::Consume, mlir::cir::MemOrder::Acquire);
-        buildAtomicCmpXchg(CGF, E, IsWeak, Dest, Ptr, Val1, Val2, Size,
-                           SuccessOrder, mlir::cir::MemOrder::Acquire, Scope);
-
-        // A failed compare-exchange is a read-only operation.  So
-        // memory_order_release and memory_order_acq_rel are not supported for
-        // the failure memory order.  They fall back to memory_order_relaxed.
-
-        // case seq_cst:
-        buildSingleMemOrderCase(builder, CaseAttrs, Regions, loc,
-                                FailureOrderVal.getType(),
-                                mlir::cir::MemOrder::SequentiallyConsistent);
-        buildAtomicCmpXchg(CGF, E, IsWeak, Dest, Ptr, Val1, Val2, Size,
-                           SuccessOrder,
-                           mlir::cir::MemOrder::SequentiallyConsistent, Scope);
-
-        os.addRegions(Regions);
-        os.addAttribute("cases", builder.getArrayAttr(CaseAttrs));
+      [&](mlir::OpBuilder &b, mlir::Location loc, mlir::OperationState &os) {
+        auto &builder = CGF.getBuilder();
+
+        mlir::Block *swtichBlock = builder.getBlock();
+
+        {
```

Comment:
We don't need all these scopes here and down, they are usually useful when using a builder insertion guard, which doesn't seem to be the case here.

Reply:
I was meant to make it clearer. But it is not bad too.